src/osd/OSD.cc

```diff
@@ -6706,6 +6719,9 @@ void OSD::handle_osd_map(MOSDMap *m)
   // even if this map isn't from a mon, we may have satisfied our subscription
   monc->sub_got("osdmap", last);
 
+  if (!m->maps.empty())
```

Comment:
@xiaoxichen i think this piece of code is repeating the `got_full_map()` if the monitor is not capping the returned MOSDMap.
```c++
  if (requested_full_first == requested_full_last) {
    dout(10) << __func__ << " " << e << ", requested " << requested_full_first
	     << ".." << requested_full_last
	     << ", now done" << dendl;
    requested_full_first = requested_full_last = 0;
```
can we reset the (requested_full_first, requested_full_last) only if the `last` is smaller than `requested_full_last`.